crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1118,6 +1118,35 @@ impl<'db> Type<'db> {
                 self_subclass_ty.is_subtype_of(db, target_subclass_ty)
             }
 
+            (Type::ClassLiteral(_), Type::Callable(_)) => {
+                let metaclass_call_symbol = self
+                    .member_lookup_with_policy(
+                        db,
+                        "__call__".into(),
+                        MemberLookupPolicy::NO_INSTANCE_FALLBACK
+                            | MemberLookupPolicy::META_CLASS_NO_TYPE_FALLBACK,
+                    )
+                    .symbol;
+
+                if let Symbol::Type(Type::BoundMethod(new_function), _) = metaclass_call_symbol {
+                    let overrides_type_call = match new_function.function(db).signature(db) {
+                        FunctionSignature::Single(signature)
+                        | FunctionSignature::Overloaded(_, Some(signature)) => {
+                            match signature.return_ty {
+                                Some(ty) => !self.is_subtype_of(db, ty.to_meta_type(db)),
+                                _ => false,
+                            }
+                        }
+                        FunctionSignature::Overloaded(..) => false,
+                    };
+                    if overrides_type_call {
```

Comment:
As discussed above, I think for now let's just always use the metaclass `__call__` signature, if present, and add a comment `// TODO: this intentionally diverges from step 1 in https://typing.python.org/en/latest/spec/constructors.html#converting-a-constructor-to-callable, which makes unwarranted assumptions`